packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -1487,11 +1482,9 @@ class StackViewport extends Viewport implements IStackViewport {
     imageId: string,
     imageIdIndex: number
   ): Promise<string> {
-    if (this.useCPURendering) {
-      await this._loadAndDisplayImageCPU(imageId, imageIdIndex);
-    } else {
-      await this._loadAndDisplayImageGPU(imageId, imageIdIndex);
-    }
+    await (this.useCPURendering
```

Comment:
You might consider assign to _loadAndDisplayImage from one of the CPU or GPU functions, and then just call it as
this._loadAndDisplayImage(iamgeId, imageIdIndex)  That way you avoid the if condition everywhere, and if in the future there is another alternate, well, it becomes easy to add.